networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -984,3 +985,84 @@ def bidirectional_dijkstra(G, source, target, weight='weight'):
                         revpath.reverse()
                         finalpath = paths[0][w] + revpath[1:]
     raise nx.NetworkXNoPath("No path between %s and %s." % (source, target))
+
+
+def johnson(G, weight='weight', new_weight=None):
+    """Compute shortest paths between all nodes in a weighted graph using
+    Johnson's algorithm.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    weight: string, optional (default='weight')
+        Edge data key corresponding to the edge weight.
+
+    new_weight: string, optional (default=None)
+        Edge data key corresponding to the new edge weight after graph transformation.
+
+    Returns
+    -------
+    distance : dictionary
+       Dictionary, keyed by source and target, of shortest paths.
+
+    Raises
+    ------
+    NetworkXError
+       If given graph is not weighted.
+
+    Examples
+    --------
+    >>> import networkx as nx
+    >>> graph = nx.DiGraph()
+    >>> graph.add_weighted_edges_from([('0', '3', 3), ('0', '1', -5), ('0', '2', 2), ('1', '2', 4),('2', '3', 1)])
```

Comment:
Line longer than 79 characters. Also, missing a space after a comma.

You can do something like:

    >>> [1, 2, 3,
    ...  4, 5, 6]